### PR TITLE
common/Finisher: batch handle perfcounter && only send signal when waiter existed.

### PR DESCRIPTION
**Pull request by majianpeng**

### CODE REVIEW

**Comment by liewegas** [comment]
rebase please?

**Comment by majianpeng** [comment]
update

File: src/common/Finisher.cc [hunk 1]
```diff
@@ -77,15 +80,13 @@ void *Finisher::finisher_thread_entry()
...
-	  end = ceph_clock_now();
-	  logger->tinc(l_finisher_complete_lat, end - start);
-	  start = end;
-        }
       }
       ldout(cct, 10) << "finisher_thread done with " << ls << dendl;
       ls.clear();
+      if (logger) {
+	logger->dec(l_finisher_queue_len, count);
+	logger->tinc(l_finisher_complete_lat, ceph_clock_now() - start);
```
Comment (liewegas) [inline src/common/Finisher.cc]: This changes what complete_lat means.. it's now a single latency measurement for the whole batch, whereas before it was a latency measurement for each completion from teh start of the batch.  Which doesn't really make any sense, so this is good!

**Comment by liewegas** [comment]
retest this please

**Pull request merged**